packages/shared/components/popups/ActivityDetailsPopup.svelte

```diff
@@ -0,0 +1,91 @@
+<script lang="typescript">
+    import { ActivityTypePill, Pill, KeyValueBlock } from 'shared/components/atoms'
+    import { Text, Button } from 'shared/components'
+    import { convertToFiat, currencies, exchangeRates, formatCurrency } from 'shared/lib/currency'
+    import { formatDate, localize } from '@core/i18n'
+    import { getOfficialExplorer } from 'shared/lib/network'
+    import { Platform } from 'shared/lib/platform'
+    import { activeProfile } from 'shared/lib/profile'
+    import { CurrencyTypes } from 'shared/lib/typings/currency'
+    import { Payload } from 'shared/lib/typings/message'
+    import { formatUnitBestMatch } from 'shared/lib/units'
+    import { getMilestoneMessageValue, wallet } from 'shared/lib/wallet'
+    import { FontWeightText } from 'shared/components/Text.svelte'
+    import { getTransactionSubjectAddressOrAccount } from '@lib/utils/transactionObject'
+
+    export let id: string
+    export let timestamp: string
+    export let confirmed: boolean
+    export let payload: Payload
+    export let balance: number // migration tx
+
+    let date = localize('error.invalidDate')
+    $: {
+        try {
+            date = formatDate(new Date(timestamp), {
+                dateStyle: 'long',
+                timeStyle: 'medium',
+            })
+        } catch {
+            date = localize('error.invalidDate')
+        }
+    }
+    const { accounts } = $wallet
+
+    const cachedMigrationTx = !payload
+    const milestonePayload = payload?.type === 'Milestone' ? payload : undefined
+    const transactionPayload = payload?.type === 'Transaction' ? payload : undefined
+
+    const explorerLink = getOfficialExplorer($accounts[0].clientOptions.network)
+
+    let value = 0
+    $: {
+        if (cachedMigrationTx) {
+            value = balance
+        } else if (milestonePayload) {
+            value = getMilestoneMessageValue(milestonePayload, $accounts)
+        } else if (transactionPayload) {
+            value = transactionPayload.data.essence.data.value
+        }
+    }
+    $: currencyValue = convertToFiat(
+        value,
+        $currencies[CurrencyTypes.USD],
+        $exchangeRates[$activeProfile?.settings.currency]
```

Comment:
Might be good to add a fallback to USD here, i.e. `$exchangeRates[$activeProfile?.settings.currency ?? CurrencyTypes.USD]`.

Reply:
Perhaps, this currently only works for IOTA and we need to rethink this anyway. Also this code already exists on develop.